File: app/src/main/java/com/dai/message/ui/music/cloud/CloudDiskAdapter.java
package com.dai.message.ui.music.cloud;

import android.databinding.DataBindingUtil;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import com.dai.message.R;
import com.dai.message.adapter.util.ViewHolder;
import com.dht.baselib.base.BaseAdapter;
import com.dht.baselib.callback.RecycleItemClickCallBack;
import com.dai.message.databinding.RecycleItemCloudDiskBinding;

import java.util.ArrayList;
import java.util.List;

/**
 * created by Administrator on 2018/12/27 16:37
 */
public class CloudDiskAdapter extends BaseAdapter<String> {

    private ArrayList<String> usernameList = new ArrayList<>();

    CloudDiskAdapter(RecycleItemClickCallBack<String> recycleItemClickCallBack) {
        this.callBack = recycleItemClickCallBack;
    }

    @Override
    public void setChangeList(List<String> mList) {
        this.mList = mList;
        notifyDataSetChanged();
    }

    @Override
    public RecyclerView.ViewHolder onCreateVH(@NonNull ViewGroup parent, int viewType) {
        RecycleItemCloudDiskBinding mBinding = DataBindingUtil.inflate(LayoutInflater.from(parent.getContext()),
                R.layout.recycle_item_cloud_disk, parent, false);
        mBinding.setCallback(callBack);
        return new ViewHolder<>(mBinding);
    }

    @SuppressWarnings("unchecked")
    @Override
    public void onBindVH(@NonNull RecyclerView.ViewHolder holder, int position) {
        if (holder instanceof ViewHolder) {
            ((ViewHolder<RecycleItemCloudDiskBinding>) holder).mBinding.tvSongName.setText(mList.get(position));
            ((ViewHolder<RecycleItemCloudDiskBinding>) holder).mBinding.tvUsername.setText(usernameList.get(position));
            ((ViewHolder<RecycleItemCloudDiskBinding>) holder).mBinding.setName(mList.get(position));
            ((ViewHolder<RecycleItemCloudDiskBinding>) holder).mBinding.setIndex(position);
        }
    }

    void setUsernameList(ArrayList<String> usernameList) {
        this.usernameList = usernameList;
    }

}

File: music/src/main/java/com/dht/music/cloud/CloudDiskAdapter.kt
package com.dht.music.cloud

import android.databinding.DataBindingUtil
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import com.dht.baselib.base.BaseAdapter
import com.dht.baselib.callback.RecycleItemClickCallBack
import com.dht.music.R
import com.dht.music.databinding.RecycleItemCloudDiskBinding
import com.dht.music.util.ViewHolder
import java.util.*

/**
 * created by Administrator on 2018/12/27 16:37
 *
 * @author Administrator
 */
class CloudDiskAdapter internal constructor(recycleItemClickCallBack: RecycleItemClickCallBack<String?>?) : BaseAdapter<String?>() {
    private var usernameList = ArrayList<String>()
    override fun setChangeList(mList: MutableList<String?>?) {
        this.mList = mList
        notifyDataSetChanged()
    }

    override fun onCreateVH(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val mBinding = DataBindingUtil.inflate<RecycleItemCloudDiskBinding>(LayoutInflater.from(parent.context),
                R.layout.recycle_item_cloud_disk, parent, false)
        mBinding.callback = callBack
        return ViewHolder(mBinding)
    }

    override fun onBindVH(holder: RecyclerView.ViewHolder, position: Int) {
        if (holder is ViewHolder<*>) {
            (holder as ViewHolder<RecycleItemCloudDiskBinding>).mBinding.tvSongName.text = mList[position]
            holder.mBinding.tvUsername.text = usernameList[position]
            holder.mBinding.name = mList[position]
            holder.mBinding.index = position
        }
    }

    fun setUsernameList(usernameList: ArrayList<String>) {
        this.usernameList = usernameList
    }

    companion object {
        private const val TAG = "dht"
    }

    init {
        callBack = recycleItemClickCallBack!!
    }


}